src/main/groovy/com/ofg/twitter/place/extractor/WeatherClient.groovy

```diff
@@ -15,20 +19,36 @@ class WeatherClient {
     }
 
     String findCity(double latitude, double longitude) {
-        return serviceRestClient.forExternalService()
-                .get()
-                .onUrl(cityFindingServiceUrl + "?lat=${latitude}&lon=${longitude}")
-                .anObject()
-                .ofType(String)
+        withHystrix('findCity') {
+            return serviceRestClient.forExternalService()
+                    .get()
+                    .onUrl(cityFindingServiceUrl + "?lat=${latitude}&lon=${longitude}")
+                    .anObject()
+                    .ofType(String)
+        }
     }
 
     String isCityExistent(String cityNameToCheck) {
-        return serviceRestClient.forExternalService()
-                .get()
-                .onUrl("$cityFindingServiceUrl?q=$cityNameToCheck")
-                .anObject()
-                .ofType(String)
+        withHystrix('isCityExistent') {
+            return serviceRestClient.forExternalService()
+                    .get()
+                    .onUrl("$cityFindingServiceUrl?q=$cityNameToCheck")
+                    .anObject()
+                    .ofType(String)
+        }
     }
 
+    private def withHystrix(String name, Closure block) {
```

Comment:
I realized it the moment I wrote it. However such API is too limited because it does not allow you to set curcit breaker, timeouts, thread pool, etc. It works here, but is not flexible enough, IMHO.